addons/website_slides_survey/static/src/js/slides_upload.js

```diff
@@ -3,12 +3,42 @@ odoo.define('website_slides_survey.upload_modal', function (require) {
 
 var core = require('web.core');
 var _t = core._t;
-var SlidesUpload = require('website_slides.upload_modal');
 
+var SlidesUpload = require('website_slides.upload_modal');
 /**
  * Management of the new 'certification' slide_type
  */
 SlidesUpload.SlideUploadDialog.include({
+    events: _.extend({}, SlidesUpload.SlideUploadDialog.prototype.events || {}, {
+        'change .o_wslides_js_slide_certification_id': '_populateWithCertificationName'
+    }),
+    /**
+     * @override
+     */
+    willStart: function () {
+        var self = this;
+        var def = new Promise((resolve) => {
+            return self._rpc({
+                route: '/slides_survey/certification/fetch_certification_info',
+                params: {
+                        fields: ['title'],
+                    }
+            }).then(function (result) {
+                if (result.read_results.length === 0 && !result.can_create) {
+                    //If there are no existing certifications and the user does not have create rights, show him
+                    //another template with a warning
+                    self.noCertification = true;
+                }
+                resolve();
+            });
+        });
+        return Promise.all([this._super.apply(this, arguments), def]);
+    },
```

Comment:
remove this override and show the warning during _bindSelect2Dropdown. (to avoid making fetch_certification_info twice)
Maybe noCertification variable is not necessary. once you know the user will not be able to create (right after the rpc call), display the warning and disable publish buttons.

Reply:
I don't know how this can be done. 
```
 this.$('#certification_id').select2(this._select2Wrapper(_t('Certification'), false,
            function () {
                return self._rpc({
            route: '/slides_survey/certification/fetch_certification_info',
            params: {
                fields: ['title'],
            }
        });
            }, 'title')
        );
```
Is only resolved once the dropdown is clicked and opened. I need to know before the rendering what I should show. I can do the rpc call once in willstart, store the results and reuse them in _bindSelect2Dropdown but i'm not sure that's a good idea.

Reply:
I pushed what I did. So there is one call to fetch_certification_info and I removed noCertification but I still have the willstart override.

Reply:
I had the idea to fill the select2 not on click but when initializing the select2, with something like
```js
    _bindSelect2Dropdown: function () {
        this._super.apply(this, arguments);
        var self = this;

        var certifications = self._rpc({
            route: '/slides_survey/certification/fetch_certification_info',
            params: {
                fields: ['title'],
            }
        });
        certifications.then(function (result) {
            if (result.can_create) {
                // show warning
                alert('dude, you don\'t have access !!');
            }
            self.$('#certification_id').select2(self._select2Wrapper(_t('Certification'), false,
                function () {
                    return Promise.resolve(result);
                }, 'title')
            );
        });
    },
```

But the problem is that the user could (if call is super slow) be able to click on the dropdown before the return of the rpc.
So we are wondering if we could not simply do nothing.
I mean by that : if there is no certification survey and you cannot create, the user will have an empty dropdown list and he won't be able to create one. And that's it.
No warning, no button disabled, ..

Reply:
I think it is indeed easier to do nothing and nothing can go wrong that way. 
en plus, dans tous les cas, I don't think it will happen very often that someone who wants to create a certification but has no rights on survey will have an empty list of surveys.

Reply:
it's done, now it does nothing anymore :)